File: apps/ios/Barns/BarnsApp/Domain/Repositories/ProductItemRepository.swift
protocol ProductItemRepository {
    func productItems() async throws -> [ProductItem]
    func productItem(id: ProductItem.ID) async throws -> ProductItem?
    func addProductItem(_ item: ProductItem) async throws
    /// Replaces the locally stored item with the same id. Local-only; never
    /// sent to a server.
    func updateProductItem(_ item: ProductItem) async throws
}

File: apps/android/app/src/main/java/com/barns/app/domain/repository/ProductItemRepository.kt
package com.barns.app.domain.repository

import com.barns.app.domain.model.ProductItem

interface ProductItemRepository {
    suspend fun productItems(): List<ProductItem>
    suspend fun productItem(id: String): ProductItem?
    suspend fun addProductItem(item: ProductItem)

    /** Replaces the locally stored item with the same id. Local-only; never
     * sent to a server. */
    suspend fun updateProductItem(item: ProductItem)
}
